src/plugins/console/public/application/components/variables_modal.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useForm, Controller, useWatch, SubmitHandler } from 'react-hook-form';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiBasicTable,
+  EuiFieldText,
+  useGeneratedHtmlId,
+  EuiForm,
+  EuiButtonIcon,
+  type EuiBasicTableColumn,
+} from '@elastic/eui';
+
+import type { DevToolsVariable, Variables } from '../../services';
+import { generateDefaultVariable } from '../../services';
+
+interface Props {
+  onClose: () => void;
+  onSaveVariables: (newVariables: DevToolsVariable[]) => void;
+  variablesService: Variables;
+}
+
+interface FormProps {
+  variables: DevToolsVariable[];
+}
+
+export function DevToolsVariablesModal({ onClose, onSaveVariables, variablesService }: Props) {
```

Comment:
I see +2.5KB in initial bundle. Please if possible move some components into async chunks.